cinema/models.py

```diff
@@ -1,107 +1,111 @@
-            if not (1 <= ticket_attr_value <= count_attrs):
-                raise ValidationError(
-                    {
-                        ticket_attr_name: f"{ticket_attr_name} number "
-                        f"must be in available range: "
-                        f"(1, {cinema_hall_attr_name}): "
-                        f"(1, {count_attrs})"
-                    }
-                )
-
-    def __str__(self):
-        return (
-            f"{str(self.movie_session)} (row: {self.row}, seat: {self.seat})"
-        )
-
-    class Meta:
-        unique_together = ("movie_session", "row", "seat")
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.db import models
+
+
+class CinemaHall(models.Model):
+    name = models.CharField(max_length=255)
+    rows = models.IntegerField()
+    seats_in_row = models.IntegerField()
+
+    @property
+    def capacity(self) -> int:
+        return self.rows * self.seats_in_row
+
+    def __str__(self):
+        return self.name
+
+
+class Genre(models.Model):
+    name = models.CharField(max_length=255, unique=True)
+
+    def __str__(self):
+        return self.name
+
+
+class Actor(models.Model):
+    first_name = models.CharField(max_length=255)
+    last_name = models.CharField(max_length=255)
+
+    @property
+    def full_name(self):
+        return str(self)
+
+    def __str__(self):
+        return self.first_name + " " + self.last_name
+
+
+class Movie(models.Model):
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+    duration = models.IntegerField()
+    genres = models.ManyToManyField(Genre)
+    actors = models.ManyToManyField(Actor)
+
+    class Meta:
+        ordering = ["title"]
+
+    def __str__(self):
+        return self.title
+
+
+class MovieSession(models.Model):
+    show_time = models.DateTimeField()
+    movie = models.ForeignKey(Movie, on_delete=models.CASCADE)
+    cinema_hall = models.ForeignKey(CinemaHall, on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ["-show_time"]
+
+    def __str__(self):
+        return self.movie.title + " " + str(self.show_time)
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL, on_delete=models.CASCADE
+    )
+
+    def __str__(self):
+        return str(self.created_at)
+
+    class Meta:
+        ordering = ["-created_at"]
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        MovieSession, on_delete=models.CASCADE, related_name="tickets"
+    )
+    order = models.ForeignKey(
+        Order, on_delete=models.CASCADE, related_name="tickets"
+    )
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    def clean(self):
+        for ticket_attr_value, ticket_attr_name, cinema_hall_attr_name in [
+            (self.row, "row", "count_rows"),
+            (self.seat, "seat", "count_seats_in_row"),

```

Comment:
The `clean` method in the `Ticket` model references `count_rows` and `count_seats_in_row` as attributes of `CinemaHall`, but these attributes do not exist. They should be `rows` and `seats_in_row` respectively. Update the attribute names to match the actual field names in the `CinemaHall` model.